src/firecracker/src/api_server_adapter.rs

```diff
@@ -76,6 +75,7 @@ impl Subscriber for ApiServerAdapter {
                     panic!("The channel's sending half was disconnected. Cannot receive data.");
                 }
             };
+            let _ = self.api_event_fd.read();
```

Comment:
Why it's necessary to consume the API event after we read from the channel?

Reply:
We need to consume the source of the event as an ack that we processed the event. If we do not consume it, epoll_wait will exit every time and report the event leading to an epoll driven busy wait loop.

Reply:
Yes, this is clear for me. But the changes from this PR moved the `self.api_event_fd.read()` call after the `match self.from_api.try_recv()`. I do not understand why we need this. I believe that it does not matter when we consume it, before or after the channel read.

Reply:
Moving the `event.read()` _after_ the `channel.try_read()` is just paranoia for the following unlikely scenario:

> We introduce some faulty logic that also consumes the api_event from some other place (I know, near zero probability, therefore characterizing this move as paranoia). What could happen is that epoll wakes this thread up, but the event is incorrectly consumed by some theoretical faulty code on a different thread, and because the event_fd is now blocking, this function blocks indefinitely.

By moving the event_fd read after the `channel.try_read()` we still get the same behavior but at least we get the channel "spurious event" warning in the logs and that would help pin-point the problem fairly fast.

Again, this move is not necessary, it's just defensive programming and it doesn't cost anything.